src/analysis/special_functions/gamma.lean

```diff
@@ -55,19 +54,26 @@ end
 /-- Euler's integral for the `Γ` function (of a real variable `s`), defined as
 `∫ x in Ioi 0, exp (-x) * x ^ (s - 1)`.
 
-See `Gamma_integral_convergent` for a proof of the convergence of the integral for `1 ≤ s`. -/
+See `Gamma_integral_convergent` for a proof of the convergence of the integral for `0 < s`. -/
 def Gamma_integral (s : ℝ) : ℝ := ∫ x in Ioi (0:ℝ), exp (-x) * x ^ (s - 1)
 
-/-- The integral defining the `Γ` function converges for real `s` with `1 ≤ s`.
-
-This is not optimal, but the optimal bound (convergence for `0 < s`) is hard to establish with the
-results currently in the library. -/
-lemma Gamma_integral_convergent {s : ℝ} (h : 1 ≤ s) :
+/-- The integral defining the `Γ` function converges for positive real `s`. -/
+lemma Gamma_integral_convergent {s : ℝ} (h : 0 < s) :
   integrable_on (λ x:ℝ, exp (-x) * x ^ (s - 1)) (Ioi 0) :=
 begin
-  refine integrable_of_is_O_exp_neg one_half_pos _ (Gamma_integrand_is_o _ ).is_O,
-  refine continuous_on_id.neg.exp.mul (continuous_on_id.rpow_const _),
-  intros x hx, right, simpa only [sub_nonneg] using h,
+  rw [←Ioc_union_Ioi_eq_Ioi (@zero_le_one ℝ _), integrable_on_union],
+  split,
+  { rw ←integrable_on_Icc_iff_integrable_on_Ioc,
+    refine integrable_on.continuous_on_mul continuous_on_id.neg.exp _ is_compact_Icc,
+    refine (interval_integrable_iff_integrable_Icc_of_le zero_le_one).mp _,
+    exact interval_integrable_rpow' (by linarith), },
+  { refine integrable_of_is_O_exp_neg one_half_pos _ (Gamma_integrand_is_o _ ).is_O,
+    refine continuous_on_id.neg.exp.mul (continuous_on_id.rpow_const _),
+    intros x hx,
+    left,
+    rw mem_Ici at hx,
+    simp,
+    linarith, }
```

Comment:
```suggestion
    exact or.inl ((zero_lt_one : (0 : ℝ) < 1).trans_le hx).ne' }
```
You should avoid non-terminal simps. Here, the proof is simple enough that you can give it directly.